presto-hive/src/main/java/com/facebook/presto/hive/SortingFileWriter.java

```diff
@@ -193,7 +192,7 @@ private void combineFiles()
             int count = min(maxOpenTempFiles, tempFiles.size() - (maxOpenTempFiles - 1));
 
             List<TempFile> smallestFiles = IntStream.range(0, count)
-                    .mapToObj(i -> tempFiles.pollFirst())
+                    .mapToObj(i -> tempFiles.poll())
```

Comment:
`tempFiles.remove()` would be better, as empty queue is an exceptional condition

Reply:
Good point, I forgot about that method. Though we will fail here with NPE due to immutable list.

Reply:
I know. Just that NPE is slightly more confusing that NoElement... Surely by looking into the code one will know why NPE.